moveit_core/collision_detection_bullet/include/moveit/collision_detection_bullet/tesseract/basic_types.h

```diff
@@ -0,0 +1,109 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * @par
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#ifndef TESSERACT_CORE_BASIC_TYPES_H
+#define TESSERACT_CORE_BASIC_TYPES_H
+
+#include <Eigen/Core>
+#include <Eigen/Geometry>
+#include <Eigen/StdVector>
+#include <geometric_shapes/shapes.h>
+#include <unordered_map>
+#include <vector>
+#include <memory>
+#include <functional>
+#include <map>
+#include <moveit/collision_detection/collision_common.h>
+
+namespace tesseract
+{
+template <typename T>
+using AlignedVector = std::vector<T, Eigen::aligned_allocator<T>>;
+
+template <typename Key, typename Value>
+using AlignedMap = std::map<Key, Value, std::less<Key>, Eigen::aligned_allocator<std::pair<const Key, Value>>>;
+
+template <typename Key, typename Value>
+using AlignedUnorderedMap = std::unordered_map<Key, Value, std::hash<Key>, std::equal_to<Key>,
+                                               Eigen::aligned_allocator<std::pair<const Key, Value>>>;
+
+/** @brief Checks if contact is allowed or not. Returns true if yes, otherwise false.
+ *
+ * The order of strings should not matter, the function should handled by the function. */
+typedef std::function<bool(const std::string&, const std::string&, const collision_detection::AllowedCollisionMatrix*)>
+    IsContactAllowedFn;
+
+enum class CollisionObjectType
+{
+  UseShapeType = 0, /**< @brief Infer the type from the type specified in the shapes::Shape class */
+
+  // all of the following convert the meshes to custom collision objects
+  ConvexHull = 1,  /**< @brief Use the mesh in shapes::Shape but make it a convex hulls collision object (if not convex
+                      it will be converted) */
+  MultiSphere = 2, /**< @brief Use the mesh and represent it by multiple spheres collision object */
+  SDF = 3          /**< @brief Use the mesh and rpresent it by a signed distance fields collision object */
+};
+
+/// Contact test data and query results information
+struct ContactTestData
+{
+  EIGEN_MAKE_ALIGNED_OPERATOR_NEW
+
+  ContactTestData(const std::vector<std::string>& active, const double& contact_distance, const IsContactAllowedFn& fn,
+                  collision_detection::CollisionResult& res, const collision_detection::CollisionRequest& req,
+                  const collision_detection::AllowedCollisionMatrix* acm)
+    : active(active)
+    , contact_distance(contact_distance)
+    , fn(fn)
+    , acm(acm)
+    , res(res)
+    , req(req)
+    , done(false)
+    , pair_done(false)
+  {
+  }
+
+  const std::vector<std::string>& active;
```

Comment:
adapt variables for mvoeit style 
```suggestion
  const std::vector<std::string>& active_;
```

Reply:
For all member variables? There are also other parts of the code where a prefixed `m_` is used (the code which directly inherits from bullet as in bullet all members are prefixed with `m_`). Should I also change this?

Reply:
`clang-tidy --fix` does this automatically. And yes, you should adapt all MoveIt code, otherwise Travis will complain anyway.

Reply:
but for struct members there are no fixed rules?

Reply:
If its derived from a bullet class then no, we can't really change it. You see this predicament with Qt-code usage, too. We may have to add ``ignore`` rules around that code, so clang-tidy doesn't complain.

Using ``clang-tidy --fix`` as @rhaschke is the best way to go, its nice and automated :+1:

Reply:
I already ran clang-tidy (super nice tool!) but this is in a struct, and for struct there are no rules specified in the clang-tidy config. Is it common practice to use `_` postfix for struct members? Then this should be added to the clang-tidy config I think. However I see them without the `_` postfix all over the collision checking code, see [here](https://github.com/ros-planning/moveit/blob/b853c6bc48c41af7475db8cbb4d11078bb8725eb/moveit_core/collision_detection/include/moveit/collision_detection/collision_common.h#L137), [here](https://github.com/ros-planning/moveit/blob/b853c6bc48c41af7475db8cbb4d11078bb8725eb/moveit_core/collision_detection/include/moveit/collision_detection/collision_common.h#L74), [here](https://github.com/ros-planning/moveit/blob/b853c6bc48c41af7475db8cbb4d11078bb8725eb/moveit_core/collision_detection_fcl/include/moveit/collision_detection_fcl/collision_common.h#L62).